### PR TITLE
Global Access Wideners

### PR DESCRIPTION
**Pull request by shartte**
This relies on a V2 branch of the access-widener project (https://github.com/shartte/access-widener/tree/v2) and is just in draft-status here to work on both in parallel.

I believe modmuss mentioned something about comments where AWs were applied? I honestly don't know how to do that.
What could be possible is adding an additional message to loom's console output when it applies AWs to a class ("[including 5 from dependencies]" or sth. along those lines)

### CODE REVIEW

File: src/main/java/net/fabricmc/loom/configuration/accesswidener/AccessWidenerJarProcessor.java [hunk 1]
```diff
@@ -76,41 +78,52 @@ public String getId() {
...
+			// For our own mod, we include local AWs, and enable strict parsing
+			var ownModReader = new AccessWidenerReader(remappingVisitor);
+			ownModReader.setStrictMode(true);
+			ownModReader.read(modAccessWidener);
 
-				if (!validNamespaces.contains(accessWidener.getNamespace())) {
-					throw new UnsupportedOperationException(String.format("Access Widener namespace '%s' is not a valid namespace, it must be one of: '%s'", accessWidener.getNamespace(), String.join(", ", validNamespaces)));
+			// For other mods, we do not use strict mode, and also filter for global AWs
+			var globalReader = new AccessWidenerReader(new GlobalOnlyDecorator(remappingVisitor));
+			for (var entry : Constants.MOD_COMPILE_ENTRIES) {
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/loom/configuration/accesswidener/AccessWidenerJarProcessor.java]: Please dont use `var`

File: src/main/java/net/fabricmc/loom/task/RemapJarTask.java [hunk 2]
```diff
@@ -152,9 +152,9 @@ public void scheduleRemap(boolean isMainRemapTask) throws Throwable {
 						byte[] data;
 
 						try {
-							data = accessWidenerJarProcessor.getRemappedAccessWidener(remapper);
+							data = accessWidenerJarProcessor.getRemappedAccessWidener(remapper, "intermediary");
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/loom/task/RemapJarTask.java]:
```suggestion
							data = accessWidenerJarProcessor.getRemappedAccessWidener(remapper, toM);
```

File: src/main/java/net/fabricmc/loom/configuration/providers/minecraft/MinecraftMappedProvider.java [hunk 3]
```diff
@@ -129,17 +132,17 @@ private void mapMinecraftJar() throws IOException {
 		}
 	}
 
-	public TinyRemapper getTinyRemapper(String fromM, String toM) throws IOException {
+	public static TinyRemapper getTinyRemapper(LoomGradleExtension extension, String fromM, String toM) throws IOException {
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/loom/configuration/providers/minecraft/MinecraftMappedProvider.java]: Why the change to make this static and pass in the extension? You can get this instance from the extension and call it none staticailly

Comment (shartte) [reply src/main/java/net/fabricmc/loom/configuration/providers/minecraft/MinecraftMappedProvider.java]: Sadly not, this class is actually instantiated _after_ the access widener jar processor is created and setup. Since this method doesnt really use any instance state, i decided to make it (and the one for the compile classpath) static, but they should probably be moved.

File: src/main/java/net/fabricmc/loom/configuration/accesswidener/AccessWidenerJarProcessor.java (hunk 1 above)
Comment (shartte) [reply src/main/java/net/fabricmc/loom/configuration/accesswidener/AccessWidenerJarProcessor.java]: Alright, I thought I could because someone else already snuck some past you :-P

Comment (modmuss50) [reply src/main/java/net/fabricmc/loom/configuration/accesswidener/AccessWidenerJarProcessor.java]: Yeah, could have even been me tbh. Its very hard to read when they are spammed like this. Going to add a checkstyle rule to prevent them. 👍

File: src/main/java/net/fabricmc/loom/configuration/providers/minecraft/MinecraftMappedProvider.java (hunk 3 above)
Comment (modmuss50) [reply src/main/java/net/fabricmc/loom/configuration/providers/minecraft/MinecraftMappedProvider.java]: Ok, thats fine. Yes, I would move these out now 👍

**Comment by shartte** [comment]
Made the PR that this is based on to AW: https://github.com/FabricMC/access-widener/pull/9

**Review by modmuss50 — APPROVED** [review]
LGTM.

**Review by modmuss50 — APPROVED** [review]
